src/utils/config.ts

```diff
@@ -0,0 +1,66 @@
+import fs from "fs";
+import path from "path";
+
+const DEFAULT_CONFIG_PATH = path.resolve(process.cwd(), "config.json");
+
+// Helper function to parse boolean values
+function parseBooleanValue(value: any, defaultValue: boolean = false): boolean {
+    if (value === undefined || value === null) return defaultValue;
+    if (typeof value === "boolean") return value;
+    if (typeof value === "string") return value.toLowerCase() === "true";
+    return defaultValue;
+}
+
+// Try to read config from JSON file
+function loadConfigFromFile(configPath: string = DEFAULT_CONFIG_PATH) {
+    try {
+        if (!fs.existsSync(configPath)) {
+            return {}; // File doesn't exist, use defaults
+        }
+        const raw = fs.readFileSync(configPath, "utf-8");
+        return JSON.parse(raw);
+    } catch (error) {
+        console.warn(`Failed to load ${configPath}:`, error);
+        return {}; // On error, use defaults
+    }
+}
+
+// Load config once at startup
+const fileConfig = loadConfigFromFile();
```

Comment:
This is a better idea than what I originally had with the singleton - I'll keep this and maybe do just some small structural changes - overall all seems good